core/types/transaction_signing.go

```diff
@@ -78,8 +82,6 @@ func LatestSigner(config *params.ChainConfig) Signer {
 		default:
 			signer = HomesteadSigner{}
 		}
-	} else {
-		signer = HomesteadSigner{}
```

Comment:
Don't drop this clause! That will make it return `nil` instead of homestead signer when chain id is missing

Reply:
fixed :+1: